cupyx/scipy/sparse/dia.py

```diff
@@ -180,6 +181,18 @@ def tocsr(self, copy=False):
         """
         return self.tocsc().tocsr()
 
+    def diagonal(self, k=0):
+        rows, cols = self.shape
+        if k <= -rows or k >= cols:
+            raise ValueError("k exceeds matrix dimensions")
+        idx, = cupy.nonzero(self.offsets == k)
+        first_col, last_col = max(0, k), min(rows + k, cols)
+        if idx.size == 0:
+            return cupy.zeros(last_col - first_col, dtype=self.data.dtype)
+        return self.data[idx[0], first_col:last_col]
+
+    diagonal.__doc__ = base.spmatrix.diagonal.__doc__
```

Comment:
Please write the docstring instead of copying it.
Thanks!

Reply:
sure. that was the pattern used in scipy.sparse, but I am fine with duplicating it here